src/main/java/org/jboss/staxmapper/QNameProvider.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.staxmapper;
+
+import javax.xml.XMLConstants;
+import javax.xml.namespace.QName;
+
+/**
+ * Provides the qualified name of an XML element or attribute.
+ * @author Paul Ferraro
+ */
+public interface QNameProvider<N extends Namespace<N>> {
```

Comment:
This is a little bit confusing, in part because it seems reasonable on the surface, but it's not clear how a qname is being provided here, or why... does this represent a function which takes a namespace and emits a qname for it? Where does it get the namespace context to know what qname prefix(es) go with what namespaces?

Reply:
This is meant to encapsulate the components of a QName. The wiring of this association is up to the implementor.  In the simplest case, the Namespace implementation (e.g. an enum) would also implement QNameProvider, and thus additionally associate the namespace with a common local name.